WordPress/Classes/Networking/MediaServiceRemoteREST.m

```diff
@@ -187,6 +187,29 @@ - (void)updateMedia:(RemoteMedia *)media
            }];
 }
 
+- (void)deleteMedia:(RemoteMedia *)media
+            success:(void (^)())success
+            failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([media isKindOfClass:[RemoteMedia class]]);
+
+    NSString *path = [NSString stringWithFormat:@"sites/%@/media/%@/delete", self.siteID, media.mediaID];
+    NSString *requestUrl = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+
+    [self.wordPressComRestApi POST:requestUrl
+                        parameters:nil
+                           success:^(id responseObject, NSHTTPURLResponse *response) {
```

Comment:
Our delete post code doesn't seem to check the response, but I guess it couldn't hurt to do so. I've updated the remotes code to do this.